### PR TITLE
feat(validator): add flag to determine behavior in public rounds

### PR DESCRIPTION
**Pull request by kangsorang**
# Description

Added a flag to determine whether output submission behavior in public rounds is enabled or disabled.

### CODE REVIEW

File: components/validator/config.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ type Config struct {
 	RollupConfig                 *rollup.Config
 	AllowNonFinalized            bool
 	OutputSubmitterEnabled       bool
+	OutputSubmitterJoinPREnabled bool
```
Comment (Pangssu) [inline components/validator/config.go:42]: I think 
```suggestion
	OutputSubmitterAllowPublicRound bool
```
is better

File: components/validator/config.go [hunk 2]
```diff
@@ -196,6 +200,10 @@ func NewValidatorConfig(cfg CLIConfig, l log.Logger, m metrics.Metricer) (*Confi
 		return nil, errors.New("ProverRPC is required when challenger enabled, but given empty")
 	}
 
+	if !cfg.OutputSubmitterEnabled && cfg.OutputSubmitterAllowPublicRound {
+		return nil, errors.New("OutputSubmitterAllowPublicRound is meaningful when OutputSubmitterEnabled enabled")
+	}
```
Comment (seolaoh) [inline components/validator/config.go:205]: How about check this condition in `Check` function in L116?

**Review by seolaoh — APPROVED** [review]
LGTM

**Review by Pangssu — APPROVED** [review]

**Review by 0xHansLee — APPROVED** [review]
I left one comment not that important.

**Pull request merged**